discord/soundboard.py

```diff
@@ -0,0 +1,324 @@
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional
+
+from . import utils
+from .mixins import Hashable
+from .partial_emoji import PartialEmoji, _EmojiTag
+from .user import User
+from .utils import MISSING
+from .asset import Asset
+
+if TYPE_CHECKING:
+    import datetime
+    from typing import Dict, Any
+
+    from .types.soundboard import (
+        BaseSoundboardSound as BaseSoundboardSoundPayload,
+        SoundboardDefaultSound as SoundboardDefaultSoundPayload,
+        SoundboardSound as SoundboardSoundPayload,
+    )
+    from .state import ConnectionState
+    from .guild import Guild
+    from .message import EmojiInputType
+
+__all__ = ('BaseSoundboardSound', 'SoundboardDefaultSound', 'SoundboardSound')
+
+
+class BaseSoundboardSound(Hashable):
+    """Represents a generic Discord soundboard sound.
+
+    .. versionadded:: 2.4
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two sounds are equal.
+
+        .. describe:: x != y
+
+            Checks if two sounds are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the sound's hash.
+
+    Attributes
+    ------------
+    id: :class:`int`
+        The ID of the sound.
+    volume: :class:`float`
+        The volume of the sound as floating point percentage (e.g. ``1.0`` for 100%).
+    """
+
+    __slots__ = ('_state', 'id', 'volume')
+
+    def __init__(self, *, state: ConnectionState, data: BaseSoundboardSoundPayload):
+        self._state: ConnectionState = state
+        self.id: int = int(data['sound_id'])
+        self._update(data)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, self.__class__):
+            return self.id == other.id
+        return NotImplemented
+
+    def __ne__(self, other: object) -> bool:
+        return not self.__eq__(other)
+
+    def _update(self, data: BaseSoundboardSoundPayload):
+        self.volume: float = data['volume']
+
+    @property
+    def file(self) -> Asset:
+        """:class:`Asset`: Returns the sound file asset."""
+        return Asset._from_soundboard_sound(self._state, self.id)
```

Comment:
This is pretty odd, you might be better off inheriting from `_AssetMixin` similar to how `Emoji` classes do it.